src/modules/DNS.rb

```diff
@@ -691,6 +615,90 @@ def read_hostname_from_install_inf
       fqhostname
     end
 
+    # Updates /etc/sysconfig/network/dhcp
+    def update_sysconfig_dhcp
+      if dhclient_set_hostname != @dhcp_hostname || write_hostname_to_hosts != @write_hostname
+        log.info("dhcp_hostname=#{@dhcp_hostname}")
+        log.info("write_hostname=#{@write_hostname}")
+
+        # @dhcp_hostname and @wrote_hostname can currently be nil only when
+        # not present in original file. So, do not add it in such case.
+        SCR.Write(
+          path(".sysconfig.network.dhcp.DHCLIENT_SET_HOSTNAME"),
+          @dhcp_hostname ? "yes" : "no"
+        ) if !@dhcp_hostname.nil?
+        SCR.Write(
+          path(".sysconfig.network.dhcp.WRITE_HOSTNAME_TO_HOSTS"),
+          @write_hostname ? "yes" : "no"
+        ) if !@write_hostname
+        SCR.Write(path(".sysconfig.network.dhcp"), nil)
+      else
+        log.info("No update for /etc/sysconfig/network/dhcp")
+      end
+    end
+
+    # Updates system with new hostname
+    def update_hostname
+      log.info("hostname=#{@hostname}")
+      log.info("domain=#{@domain}")
+
+      # change the hostname
+      SCR.Execute(path(".target.bash"), Ops.add("/bin/hostname ", @hostname))
+
+      # build and write FQDN hostname
+      fqhostname = Hostname.MergeFQ(@hostname, @domain)
+      @oldhostname = fqhostname # #49634
+
+      SCR.Write(
+        path(".target.string"),
+        HOSTNAME_PATH,
+        Ops.add(fqhostname, "\n")
+      )
+
+      create_hostname_link
+    end
+
+    # Updates /etc/sysconfig/network/config
+    def update_sysconfig_config
+      log.info("nameservers=#{@nameservers}")
+      log.info("searchlist=#{@searchlist}")
+
+      SCR.Write(
+        path(".sysconfig.network.config.NETCONFIG_DNS_POLICY"),
+        @resolv_conf_policy
+      )
+      SCR.Write(
+        path(".sysconfig.network.config.NETCONFIG_DNS_STATIC_SEARCHLIST"),
+        Builtins.mergestring(@searchlist, " ")
+      )
+      SCR.Write(
+        path(".sysconfig.network.config.NETCONFIG_DNS_STATIC_SERVERS"),
+        Builtins.mergestring(@nameservers, " ")
+      )
+      SCR.Write(path(".sysconfig.network.config"), nil)
+
+      SCR.Execute(path(".target.bash"), "/sbin/netconfig update")
+    end
+
+    # A constant for translating sysconfig's yes/no values into boolean
+    SYSCFG_TO_BOOL = { "yes" => true, "no" => false }.freeze
+
+    # Reads value of DHCLIENT_SET_HOSTNAME and translates it to boolean
+    #
+    # return {true, false, nil} "yes" => true, "no" => false, otherwise or not
+    # present => nil
+    def dhclient_set_hostname
+      SYSCFG_TO_BOOL[SCR.Read(path(".sysconfig.network.dhcp.DHCLIENT_SET_HOSTNAME"))]
+    end
+
+    # Reads value of WRITE_HOSTNAME_TO_HOSTS and translates it to boolean
+    #
+    # return {true, false, nil} "yes" => true, "no" => false, otherwise or not
+    # present => nil
+    def write_hostname_to_hosts
```

Comment:
A method named `write_*` that in fact calls `SCR.Read`?! I know it copies the sysconfig name but here I would prefix it with `should_` or something.